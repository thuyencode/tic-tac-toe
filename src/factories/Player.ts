import { Mark } from '../libs/types'

/**
 * Creates a new player object with the given player mark.
 *
 * @param {Mark} playerMark - The mark of the player.
 * @return {Object} - The player object with the mark, getScore, addOneScore, and resetScore methods.
 */
export function Player(mark: Mark) {
  let score = 0

  /**
   * Returns the current score.
   */
  const getScore = () => score
  const addOneScore = () => {
    score++
  }
  /**
   * Resets the score to 0.
   */
  const resetScore = () => {
    score = 0
  }

  return { mark, getScore, addOneScore, resetScore }
}
