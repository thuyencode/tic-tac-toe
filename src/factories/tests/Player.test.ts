import { describe, expect, test } from 'bun:test'
import { Player } from '../Player'

describe('Unit tests for Player.ts', () => {
  const MARK = 'X'

  test(`Player's score can be changed`, () => {
    const player = Player(MARK)

    expect(player.getScore()).toEqual(0)

    player.addOneScore()
    expect(player.getScore()).toEqual(1)

    player.resetScore()
    expect(player.getScore()).toEqual(0)
  })
})
