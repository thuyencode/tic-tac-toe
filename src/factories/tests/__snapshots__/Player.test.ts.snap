// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`math addition 1`] = `20`;
